### PR TITLE
Add and use generics.is_empty() and generics.is_own_empty, rather than using generics' attributes

### PR DESCRIPTION
**Pull request by spastorino**
r? @compiler-errors 

Related to #123929

### CODE REVIEW

File: src/tools/rustfmt/src/items.rs [hunk 1]
```diff
@@ -1725,7 +1725,7 @@ fn rewrite_ty<R: Rewrite>(
     result.push_str(&format!("{}type ", format_visibility(context, vis)));
     let ident_str = rewrite_ident(context, ident);
 
-    if generics.params.is_empty() {
+    if generics.is_empty() {
```
Comment (compiler-errors) [inline src/tools/rustfmt/src/items.rs:1728]: Please revert this

**Review by compiler-errors — CHANGES_REQUESTED** [review]
Please fix clippy and rustfmt, otherwise looks fine. I will review again when it's fixed.

File: src/tools/rustfmt/src/items.rs (hunk 1 above)
Comment (spastorino) [reply src/tools/rustfmt/src/items.rs:1728]: Ohh just figured that this is `ast::Generics`

**Comment by spastorino** [comment]
> Please fix clippy and rustfmt, otherwise looks fine. I will review again when it's fixed.

Not sure what do you mean by fix clippy. About rustfmt is fixed, didn't realize rustfmt was using `ast::Generics`.

**Review by compiler-errors — APPROVED** [review]

**Pull request merged**